components/alert/alert.js

```diff
@@ -0,0 +1,153 @@
+import '../button/button-icon.js';
+import '../button/button-subtle.js';
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { bodyCompactStyles } from '../typography/styles.js';
+import { LocalizeStaticMixin } from '../../mixins/localize-static-mixin.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+class Alert extends LocalizeStaticMixin(RtlMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			buttonText: { type: String, attribute: 'button-text' },
+			hasCloseButton: { type: Boolean, attribute: 'has-close-button' },
+			subtext: { type: String },
+			type: { type: String, reflect: true }
+		};
+	}
+	static get styles() {
+		return [bodyCompactStyles, css`
+
+			:host {
+				animation: 600ms ease drop-in;
+				background: white;
+				border-radius: 0.3rem;
+				border: 1px solid var(--d2l-color-mica);
+				box-sizing: border-box;
+				display: flex;
+				flex: 1;
+				max-width: 710px;
+				width: 100%;
+			}
+
+			:host([hidden]) {
+				display: none;
+			}
+
+			.d2l-alert-highlight {
+				border-radius: 0.3rem 0 0 0.3rem;
+				bottom: 0;
+				left: 0;
+				margin: -1px;
+				min-width: 0.3rem;
+				top: 0;
+				width: 0.3rem;
+			}
+			:host([dir="rtl"]) .d2l-alert-highlight {
+				border-radius: 0 0.3rem 0.3rem 0;
+				left: auto;
+				right: 0;
+			}
+			:host([type="critical"]) .d2l-alert-highlight,
+			:host([type="error"]) .d2l-alert-highlight {
+				background-color: var(--d2l-color-feedback-error);
+			}
+			:host([type="warning"]) .d2l-alert-highlight {
+				background-color: var(--d2l-color-feedback-warning);
+			}
+			:host([type="default"]) .d2l-alert-highlight,
+			:host([type="call-to-action"]) .d2l-alert-highlight {
+				background-color: var(--d2l-color-feedback-action);
+			}
+			:host([type="success"]) .d2l-alert-highlight {
+				background-color: var(--d2l-color-feedback-success);
+			}
+
+			.d2l-alert-text {
+				flex: 1;
+				padding: 1rem 1.2rem 1rem 1.5rem;
+				position: relative;
+			}
+			:host([dir="rtl"]) .d2l-alert-text {
+				padding-left: 1.2rem;
+				padding-right: 1.5rem;
+			}
+			.d2l-alert-subtext {
+				margin: 0.5rem 0 0;
+			}
+
+			.d2l-alert-action {
+				margin: 0.3rem;
+			}
+
+			@keyframes drop-in {
```

Comment:
I know this was copied forward from the original.  I think think could have been done with a simple transition though.

Nit: I think the `0.0` on opacity below could just be `0`.  Another nit: the translate could have been `translateY(...)`.  Meh.